src/encrypt/des_ecb.c

```diff
@@ -0,0 +1,185 @@
+
+    rv = funcs->C_EncryptInit(session, &mech, des_key);
+    if (CKR_OK != rv) {
+        printf("Encryption Init failed: %lu\n", rv);
+        return rv;
+    }
+
+    // Determine how much memory will be required to hold the ciphertext.
+    rv = funcs->C_Encrypt(session, plaintext, plaintext_length, NULL, &ciphertext_length);
+    if (CKR_OK != rv) {
+        printf("Encryption failed: %lu\n", rv);
+        return rv;
+    }
+
+    // Allocate the required memory.
+    CK_BYTE_PTR ciphertext = malloc(ciphertext_length);
+    if (NULL == ciphertext) {
+        printf("Could not allocate memory for ciphertext\n");
+        return rv;
+    }
+    memset(ciphertext, 0, ciphertext_length);
+
+    // Encrypt the data.
+    rv = funcs->C_Encrypt(session, plaintext, plaintext_length, ciphertext, &ciphertext_length);
+    if (CKR_OK != rv) {
+        printf("Encryption failed: %lu\n", rv);
+        goto done;
+    }
+
+    // Print just the ciphertext in hex format
+    printf("Ciphertext: ");
+    print_bytes_as_hex(ciphertext, ciphertext_length);
+    printf("Ciphertext length: %lu\n", ciphertext_length);
+
+    //**********************************************************************************************
+    // Decrypt
+    //********************************************************************************************** 
+
+    rv = funcs->C_DecryptInit(session, &mech, des_key);
+    if (CKR_OK != rv) {
+        printf("Decryption Init failed: %lu\n", rv);
+        return rv;
+    }
+
+    // Determine how much memory is required to hold the decrypted text.
+    CK_BYTE_PTR decrypted_ciphertext = NULL;
+    CK_ULONG decrypted_ciphertext_length = 0;
+    rv = funcs->C_Decrypt(session, ciphertext, ciphertext_length, NULL, &decrypted_ciphertext_length);
+    if (CKR_OK != rv) {
+        printf("Decryption failed: %lu\n", rv);
+        goto done;
+    }
+
+    // Allocate memory for the decrypted ciphertext.
+    decrypted_ciphertext = malloc(decrypted_ciphertext_length);
+    if (NULL == decrypted_ciphertext) {
+        rv = 1;
+        printf("Could not allocate memory for decrypted ciphertext\n");
+        goto done;
+    }
+
+    // Decrypt the ciphertext.
+    rv = funcs->C_Decrypt(session, ciphertext, ciphertext_length, decrypted_ciphertext, &decrypted_ciphertext_length);
+    if (CKR_OK != rv) {
+        printf("Decryption failed: %lu\n", rv);
+        goto done;
+    }
+
+    printf("Decrypted ciphertext: %.*s\n", (int)decrypted_ciphertext_length, decrypted_ciphertext);
+    printf("Decrypted ciphertext length: %lu\n", decrypted_ciphertext_length);
+
+done:
+    if (NULL != decrypted_ciphertext) {
+        free(decrypted_ciphertext);
+    }
+
+    if (NULL != ciphertext) {
+        free(ciphertext);
+    }
+    return rv;
+}
+
+int main(int argc, char **argv) {
+    CK_RV rv;
+    CK_SESSION_HANDLE session;
+
+    struct pkcs_arguments args = {};
+    if (get_pkcs_args(argc, argv, &args) < 0) {
+        return 1;
```

Comment:
And the returns to be `EXIT_FAILURE` or `EXIT_SUCCESS`?

Reply:
resolved.